### PR TITLE
feat(stream watchers) end streaming if too many notifications

### PR DESCRIPTION
**Pull request by cenkalti**

### CODE REVIEW

File: store/watcher_hub.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ func (wh *watcherHub) watch(key string, recursive, stream bool, index uint64) (*
 	}
 
 	w := &Watcher{
-		EventChan:  make(chan *Event, 1), // use a buffered channel
+		// use a big buffered channel to deal with bursts in notifications
+		EventChan:  make(chan *Event, wh.EventHistory.Queue.Capacity),
```
Comment (xiang90) [inline store/watcher_hub.go]: @cenkalti Please do not do this. This will use a lot memory. And I do not think stream watcher is our main use case.

Comment (cenkalti) [inline store/watcher_hub.go]: OK. I will remove it.

**Comment by cenkalti** [comment]
I have removed the code that icreased the buffer size of event channel.

File: store/watcher.go [hunk 2]
```diff
@@ -51,10 +51,8 @@ func (w *Watcher) notify(e *Event, originalPath bool, deleted bool) bool {
 		// but we cannot block here. blocking will lead the whole etcd system to hang.
 		// create a go-routine to handle the blocking case
```
Comment (xiang90) [inline store/watcher.go]: can you change the comment accordingly? Thanks.

**Comment by cenkalti** [comment]
Updated the comment in `Watcher.notify()`.

**Comment by xiang90** [comment]
@cenkalti Thanks!

**Pull request merged**